lib/web/recordingplayback_test.go

```diff
@@ -0,0 +1,723 @@
+				startOffset:          0,
+				endOffset:            0,
+				requestID:            0,
+				requestCurrentScreen: false,
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := decodeBinaryRequest(tt.data)
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func TestEncodeScreenEvent(t *testing.T) {
+	state := vt10x.TerminalState{
+		Cols: 80,
+		Rows: 24,
+	}
+	cols := 80
+	rows := 24
+	cursor := vt10x.Cursor{X: 10, Y: 5}
+
+	result := encodeScreenEvent(state, cols, rows, cursor)
+
+	require.Greater(t, len(result), requestHeaderSize)
+	require.Equal(t, eventTypeScreen, result[0])
+	require.Equal(t, uint32(cols), binary.BigEndian.Uint32(result[1:5]))
+	require.Equal(t, uint32(rows), binary.BigEndian.Uint32(result[5:9]))
+	require.Equal(t, uint32(cursor.X), binary.BigEndian.Uint32(result[9:13]))
+	require.Equal(t, uint32(cursor.Y), binary.BigEndian.Uint32(result[13:17]))
+}
+
+func TestResizeTerminalEvent(t *testing.T) {
+	tests := []struct {
+		name         string
+		sizeString   string
+		expectResize bool
+	}{
+		{
+			name:         "valid size string",
+			sizeString:   "80:24",
+			expectResize: true,
+		},
+		{
+			name:         "invalid format - no colon",
+			sizeString:   "8024",
+			expectResize: false,
+		},
+		{
+			name:         "invalid format - too many parts",
+			sizeString:   "80:24:10",
+			expectResize: false,
+		},
+		{
+			name:         "invalid cols",
+			sizeString:   "abc:24",
+			expectResize: false,
+		},
+		{
+			name:         "invalid rows",
+			sizeString:   "80:xyz",
+			expectResize: false,
+		},
+		{
+			name:         "empty string",
+			sizeString:   "",
+			expectResize: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			s := &recordingPlayback{
+				ctx:    ctx,
+				logger: slog.Default(),
+			}
+			s.terminal.vt = vt10x.New()
+
+			initialCols, initialRows := s.terminal.vt.Size()
+
+			s.resizeTerminal(tt.sizeString)
+
+			if tt.expectResize {
+				parts := strings.Split(tt.sizeString, ":")
+				expectedCols, _ := strconv.Atoi(parts[0])
+				expectedRows, _ := strconv.Atoi(parts[1])
+
+				newCols, newRows := s.terminal.vt.Size()
+				require.Equal(t, expectedCols, newCols)
+				require.Equal(t, expectedRows, newRows)
+			} else {
+				newCols, newRows := s.terminal.vt.Size()
+
+				require.Equal(t, initialCols, newCols)
+				require.Equal(t, initialRows, newRows)
+			}
+		})
+	}
+}
+
+func TestCreateTaskContext(t *testing.T) {
+	ctx := context.Background()
+	s := &recordingPlayback{
+		ctx:    ctx,
+		logger: slog.Default(),
+	}
+
+	// Create first task
+	taskCtx1 := s.createTaskContext()
+	require.NotNil(t, taskCtx1)
+	require.NotNil(t, s.activeTask)
+
+	select {
+	case <-taskCtx1.Done():
+		t.Fatal("Task context should not be cancelled yet")
+	default:
+		// Expected
+	}
```

Comment:
```suggestion
	require.NoError(t, taskCtx1.Err(), "task context should not be canceled yet")
```

Feels a little easier to comprehend than the select, and works because:

> // If Done is not yet closed, Err returns nil.
> // If Done is closed, Err returns a non-nil error explaining why: